src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/OpportunitiesOrchestrator.cs

```diff
@@ -134,12 +160,73 @@ public async Task<ApplyViewModel> GetApplyViewModel(ApplicationRequest request)
                 Sectors = "-",
                 Locations = "-",
                 MoreDetail = "-",
-                ContactName = "-",
-                EmailAddress = "-",
-                WebsiteUrl = "-"
+                ContactName = string.IsNullOrWhiteSpace(contactName) ? "-" : contactName,
+                EmailAddresses = emailAddresses,
+                WebsiteUrl = string.IsNullOrEmpty(application.BusinessWebsite) ? "-" : application.BusinessWebsite,
             };
         }
 
+        public async Task<ContactDetailsViewModel> GetContactDetailsViewModel(ContactDetailsRequest contactDetailsRequest)
+        {
+            var opportunityDto = await _opportunitiesService.GetOpportunity(contactDetailsRequest.PledgeId);
+
+            if (opportunityDto == null)
+            {
+                return null;
+            }
+
+            var encodedPledgeId = _encodingService.Encode(opportunityDto.Id, EncodingType.PledgeId);
+
+            var opportunitySummaryViewModel = await GetOpportunitySummaryViewModel(opportunityDto, encodedPledgeId);
```

Comment:
This method depends upon that tagservice, which is chatty with the api and is obsolete, reflecting the original approach. You'll need to add a new method for creating the summary viewmodel and use that here instead (leave the current one for other usages). The new method probably just needs to accept a set of primitive parameters (name, anonymity bool, amount, ref data items, etc.)

Reply:
This has been addressed while looking at the above issue.